### ISSUE TITLE
Using the rancher2_cluster resource results in invalid RKE configuration

### ISSUE THREAD

**Issue by MG2R** [rancher/terraform-provider-rancher2#410]
I've been trying to set up a Kubernetes cluster defined with Terraform. This is the relevant Terraform code:

```tf
resource "rancher2_cluster" "pocdebug" {
  name = "pocdebug"
  description = "Used for testing changes without affecting active dev environments."

  # Kubernetes options
  rke_config {
    kubernetes_version = "v1.18.3-rancher2-2"
    network {
      plugin = "canal"
    }
  }
  enable_network_policy = false
  enable_cluster_monitoring = false
}
```

Whenever I try to add nodes to this cluster using the join command, Rancher would give me the following error in the UI:

> Can not find RKE state file: open /var/lib/rancher/management-state/rke/rke-865422716/cluster.rkestate: no such file or directory

Googling that seemed to indicate that my node names were wrong, but that wasn't the case. After some digging in the API output, I discovered that the issue lies in the `rancherKubernetesEngineConfig` that gets generated. I contains the following but:

```json
rancherKubernetesEngineConfig": {
    <...>
    "clusterName": "pocdebug",
    <...>
}
```

According to the documentation, that argument cannot be passed to that `rancher2_cluster` TF resource. It doesn't show up in the GUI, unless I "edit as YAML" the cluster configuration. Removing the offending line ( highlighted in screenshot) allows me to successfully add nodes to the cluster. However, that requires manual intervention in the GUI. This breaks my automated, config-as-code workflow.
![tempsnip](https://user-images.githubusercontent.com/5439592/89515222-1cd6c880-d7d7-11ea-863e-f0537b7b4bdf.png)

Is there a way to reliably work around this issue?

Verified this issue on both v1.9.0 and v1.10.0 of the provider. Using Terraform 0.12.28.

**Comment by rawmind0** [comment]
@MG2R , `cluster_name` under `rancherKubernetesEngineConfig` is equal to `name` set at `rancher2_cluster` resource This is a valid field on `rancherKubernetesEngineConfig` . Seems weird, i don't think this is causing your issue. I'm not able to reproduce your issue, i've just copied and pasted your terraform code, creating a cluster and adding nodes to it without any issue. 

I guess you issue may be related to https://github.com/rancher/rancher/issues/25376 . Have you checked that you are not using any non-valid value at node-command ??

**Comment by MG2R** [comment]
Hey. Sorry I forgot to update this. I figured out just an hour ago. I'm using Puppet to run the Rancher agent docker container on cluster members. The puppetlabs/docker module has a bug which prevented the containers to be attached to the host network. Just submitted a PR to fix that but hadn't gotten around to updating here.

I thought the problem outlined above was the root cause because the moment I did that workaround (removing the "offending" param) was also the moment I manually ran the docker command. Later when I let Puppet do the work again, it didn't work anymore.

I should've noticed the IP addresses for the nodes as reported by Rancher were all bogus. Sorry for the confusion here.
